packages/web/src/components/admin_page/feature_switch_page/render_switch.jsx

```diff
@@ -0,0 +1,178 @@
+import FeatureRow from './row_switch';
+import { FsContext } from '../../../feature_switches';
+import toast from 'react-hot-toast';
+import styles from './featureSwitch.module.css';
+
+const UpdateSwitch = ({ onUpdate }) => {
+    const [newSwitch, setNewSwitch] = useState({
+        switch_id: '',
+        is_enabled: true,
+    });
+
+    const handleChange = e => {
+        const { name, value } = e.target;
+        setNewSwitch(prevState => ({
+            ...prevState,
+            [name]: name === 'is_enabled' ? value === 'true' : value,
+        }));
+    };
+
+    const handleSubmit = e => {
+        e.preventDefault();
+        const trimmedSwitchId = newSwitch.switch_id.trim();
+
+        if (trimmedSwitchId === '') {
+            toast.error('Please enter switch name');
+        } else if (trimmedSwitchId.length > 10) {
+            toast.error('Switch name must be 10 characters or less');
+        } else {
+            newSwitch.switch_id = trimmedSwitchId;
+            onUpdate(newSwitch);
+        }
+    };
+
+    return (
+        <Form onSubmit={handleSubmit}>
+            <Form.Group className="mb-3" controlId="formBasicEmail">
+                <Form.Label>Set a feature:</Form.Label>
+                <Form.Control
+                    type="text"
+                    placeholder="Enter new feature switch"
+                    onChange={handleChange}
+                    name="switch_id"
+                    value={newSwitch.switch_id}
+                />
+                <Form.Text className="text-muted">Add a new feature</Form.Text>
+            </Form.Group>
+            <Form.Group className="mb-3" controlId="formBasicIsEnabled">
+                <Form.Label>Set it to true or false:</Form.Label>
+                <Form.Select onChange={handleChange} name="is_enabled" value={newSwitch.is_enabled.toString()}>
+                    <option value="true">True</option>
+                    <option value="false">False</option>
+                </Form.Select>
+                <Form.Text className="text-muted"></Form.Text>
+            </Form.Group>
+
+            <Button type="submit" variant="primary">
+                Add
+            </Button>
+        </Form>
+    );
+};
+
+export const popover = (onUpdate, errorMessage) => {
+    console.log(errorMessage);
+    return (
+        <Popover id="popover-basic">
+            <Popover.Header as="h3">Want to update or remove?</Popover.Header>
+            <Popover.Body>
+                {errorMessage ? <Alert variant="warning">{errorMessage}</Alert> : null}
+                <UpdateSwitch onUpdate={onUpdate} />
+            </Popover.Body>
+        </Popover>
+    );
+};
+
+// Creates a new object updateSwitches that makes a copy of switches and updates a specific switch's is_enabled property
+const updateSwitches = (switches, setSwitches) => data => {
+    setSwitches({ ...switches, [data.switch_id]: data.is_enabled });
```

Comment:
What does updateSwitches refer to here? There's no such object.


Additionally, this function is used for inserting new switches as well.
Make sure you test your comments in chatgpt for clarity. If chatgpt is unable to properly tell what your code is doing by only reading the comment, your description is not clear enough or is incorrect.